subsys/bluetooth/host/hci_core.c

```diff
@@ -746,13 +746,23 @@ static void le_conn_complete(struct net_buf *buf)
 		return;
 	}
 
-	id_addr = find_id_addr(&evt->peer_addr);
+	bt_addr_le_copy(&id_addr, &evt->peer_addr);
+
+	/* Translate "enhanced" identity address type to normal one */
+	if (id_addr.type == BT_ADDR_LE_PUBLIC_ID ||
+	    id_addr.type == BT_ADDR_LE_RANDOM_ID) {
+		id_addr.type -= BT_ADDR_LE_PUBLIC_ID;
```

Comment:
I remember you commenting on using if (addr_type >= ID_PUBLIC) during the LL privacy review. Isn't this similar?

Reply:
I'm not completely sure what you mean, but I think it's better to do explicit comparisons instead of the more generic >= to make it clear that it's these specific two values we're interested in.

Reply:
Well I was referring to the id_addr.type -= BT_ADDR_LE_PUBLIC_ID, which relies on the particular sequence of the address types.

Reply:
Right. however the sequence is defined in the HCI spec and will never change, so I'd say we're good here. It'd be different if these were self-made-up values coming from an enum or similar.